scripts/_common.sh

```diff
@@ -4,10 +4,81 @@
 # COMMON VARIABLES
 #=================================================
 
+# dependencies used by the app
+pkg_dependencies="postgresql postgresql-contrib pgloader"
+
 #=================================================
 # PERSONAL HELPERS
 #=================================================
 
+mariadb-to-pg() {
+
+        ynh_print_info --message="Migrating to PostgreSQL database..."
+
+        mysqlpwd=$(ynh_app_setting_get --app=$app --key=mysqlpwd)
+
+        # In old instance db_user is `mmuser`
+        mysql_db_user="$db_user"
+        if ynh_mysql_connect_as --user="mmuser" --password="$mysqlpwd" 2> /dev/null <<< ";"; then
+            mysql_db_user="mmuser"
+        fi
+
+        # Initialize PostgreSQL database
+        ynh_psql_test_if_first_run
+        ynh_psql_setup_db --db_user=$db_user --db_name=$db_name --db_pwd=$mysqlpwd
+        psqlpwd=$(ynh_app_setting_get --app=$app --key=psqlpwd)
+
+        # Configure the new database and run Mattermost in order to create tables
+        ynh_write_var_in_file --file="$final_path/config/config.json" --key="DriverName" --value="postgres" --after="SqlSettings"
+        ynh_write_var_in_file --file="$final_path/config/config.json" --key="DataSource" --value="postgres://$db_user:$psqlpwd@localhost:5432/$db_name?sslmode=disable&connect_timeout=10" --after="SqlSettings"
+        cat "$final_path/config/config.json"
+        pushd $final_path
+        ynh_systemd_action --service_name="$app" --action="stop"
+        set +e
+		sudo -u mattermost timeout --preserve-status 30 "./bin/mattermost" 
```

Comment:
On my (slow) server 30s was too short to start the server completely, and the upgrade failed when this step timed out. I would recommend 2 minutes instead (to account for slow rasperry pis, etc)

```suggestion
		sudo -u mattermost timeout --preserve-status 120 "./bin/mattermost" 
```